### PR TITLE
[dev2] Configuration files: global.conf

### PR DESCRIPTION
**Pull request by franramirez688**
Coming from https://github.com/conan-io/docs/pull/2947

### CODE REVIEW

File: reference/config_files/global_conf.rst [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    tools.microsoft.msbuild:verbosity=Diagnostic
+    tools.microsoft.msbuild:max_cpu_count=2
+    tools.microsoft.msbuild:vs_version = 16
+    tools.build:jobs=10
+
+
+Configuration file template
+---------------------------
+
+Available since: `1.46.0 <https://github.com/conan-io/conan/releases/tag/1.46.0>`_
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:135]:
```suggestion
```

File: reference/config_files/global_conf.rst [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+
+Tools configurations
+--------------------
+
+Tools and user configurations allows them to be defined both in the *global.conf* file and in profile files. Profile values will
+have priority over globally defined ones in *global.conf*, and can be defined as:
+
+.. code-block:: text
+
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:121]:
```suggestion
.. code-block:: text
    :caption: profile (not global.conf)

```

File: reference/config_files/global_conf.rst [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+     # Using all the cores automatically
+     tools.build:jobs={{os.cpu_count()}}
+     # Using the current OS
+     user.myconf.system:name = {{platform.system()}}
+
+
+.. note::
+
+    The Python packages passed to render the template are ``os`` and ``platform`` for all platforms and ``distro`` in Linux platforms.
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:150]:
```suggestion
The Python packages passed to render the template are ``os`` and ``platform`` for all platforms and ``distro`` in Linux platforms.
```
Not worth a note, not that relevant, normal text is maybe ok?

File: reference/config_files/global_conf.rst [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
+
+Configuration data types
+------------------------
+
+Available since: `1.46.0 <https://github.com/conan-io/conan/releases/tag/1.46.0>`_
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:156]:
```suggestion
```

File: reference/config_files/global_conf.rst [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+    # List of values
+    user.myconf.build:ldflags=["--flag1", "--flag2"]
+    # Dictionary
+    tools.microsoft.msbuildtoolchain:compile_options={"ExceptionHandling": "Async"}
+
+
+Configuration data operators
+----------------------------
+
+Available since: `1.46.0 <https://github.com/conan-io/conan/releases/tag/1.46.0>`_
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:177]:
```suggestion
```

File: reference/config_files/global_conf.rst [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+    user.myconf.build:flags+=["-f2"]
+
+    # Prepend the value ["-f0"] => ["-f0", "-f1", "-f2"]
+    user.myconf.build:flags=+["-f0"]
+
+    # Unset the value
+    user.myconf.build:flags=!
+
+
+Configuration in your profiles
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:206]: Maybe move this section to the "profiles" section, and link it from here.

File: reference/config_files/global_conf.rst [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+    *:tools.cmake.cmaketoolchain:generator=Ninja
+
+The result is that you're specifying a general `generator` for all your packages, and that's it. The `zlib` line has no
+effect because it's the first one evaluated, and after that, Conan is overriding that specific pattern with the most
+general one, so it deserves to pay special attention to the order.
+
+
+.. _conf_in_recipes:
+
+Configuration in your recipes
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:279]: Not belonging here?

File: reference/config_files/global_conf.rst [hunk 8]
```diff
@@ -0,0 +1,383 @@
...
+            self.conf_info.remove("user.myconf.build:ldflags", "--flag1")  # == ["--flag0", "--flag2", "--flag3"]
+            # Removing completely the configuration
+            self.conf_info.pop("tools.system.package_manager:sudo")
+
+
+.. important::
+
+    Legacy configuration methods to set/get values like ``self.conf_info["xxxxx"] = "yyyyy"`` and ``v = self.conf_info["xxxxx"]`` are
+    deprecated since Conan 1.46 version. Use ``self.conf_info.define("xxxxx", "yyyyy")`` and ``v = self.conf_info.get("xxxxx")`` instead
+    like the example above.
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:342]:
```suggestion
```

File: reference/config_files/global_conf.rst [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+
+
+Configuration from build_requires
```
Comment (memsharded) [inline reference/config_files/global_conf.rst:345]: Same, this is not global.conf reference?

File: reference/config_files/global_conf.rst (hunk 6 above)
Comment (franramirez688) [reply reference/config_files/global_conf.rst:202]: Let's do it when adding the `profiles` section.

File: reference/config_files/global_conf.rst [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+.. important::
+
+    Remember to run that command locally to actually see the latest list because this one may be outdated.
+
+
+Tools configurations
+--------------------
+
+Tools and user configurations allows them to be defined both in the *global.conf* file and in profile files. Profile values will
```
Comment (RubenRBS) [inline reference/config_files/global_conf.rst:117]:
```suggestion
Tools and user configurations allow them to be defined both in the *global.conf* file and in profile files. Profile values will
```

File: reference/config_files/global_conf.rst [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+    tools.microsoft.msbuild:max_cpu_count=2
+    tools.microsoft.msbuild:vs_version = 16
+    tools.build:jobs=10
+
+
+Configuration file template
+---------------------------
+
+
+It is possible to use **jinja2** template engine for *global.conf*. When Conan loads this file, immediately parses
```
Comment (RubenRBS) [inline reference/config_files/global_conf.rst:137]:
```suggestion
It is possible to use **jinja2** template engine for *global.conf*. When Conan loads this file, it immediately parses
```

File: reference/config_files/global_conf.rst [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+
+You can use package patterns to apply the configuration in those dependencies which are matching:
+
+.. code-block:: text
+
+    *:tools.cmake.cmaketoolchain:generator=Ninja
+    zlib:tools.cmake.cmaketoolchain:generator=Visual Studio 16 2019
+
+This example shows you how to specify a general `generator` for all your packages, but for `zlib` one. `zlib` is defining
+`Visual Studio 16 2019` as its own generator.
```
Comment (RubenRBS) [inline reference/config_files/global_conf.rst:258]:
```suggestion
This example shows you how to specify a general `generator` for all your packages, except for for `zlib`, which is defining
`Visual Studio 16 2019` as its own generator.
```

**Review by RubenRBS — APPROVED** [review]
👍

File: reference/config_files/global_conf.rst [hunk 13]
```diff
@@ -0,0 +1,293 @@
...
+    tools.microsoft.msbuildtoolchain:compile_options: Dictionary with MSBuild compiler options
+    tools.system.package_manager:mode: Mode for package_manager tools: 'check' or 'install'
+    tools.system.package_manager:sudo: Use 'sudo' when invoking the package manager tools in Linux (False by default)
+    tools.system.package_manager:sudo_askpass: Use the '-A' argument if using sudo in Linux to invoke the system package manager (False by default)
+    tools.system.package_manager:tool: Default package manager tool: 'apt-get', 'yum', 'dnf', 'brew', 'pacman', 'choco', 'zypper', 'pkg' or 'pkgutil'
+
+
+.. important::
+
+    Remember to run that command locally to actually see the latest list because this one may be outdated.
```
Comment (czoido) [inline reference/config_files/global_conf.rst:111]:
```suggestion
```
I would not warn about this, this list should never be outdated

File: reference/config_files/global_conf.rst (hunk 12 above)
Comment (franramirez688) [reply reference/config_files/global_conf.rst:258]:
```suggestion
This example shows you how to specify a general ``generator`` for all your packages except for `zlib` which is defining
`Visual Studio 16 2019` as its generator.
```

**Review by memsharded — APPROVED** [review]

File: reference/config_files/global_conf.rst [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+
+global.conf
+===========
+
+The **global.conf** file is located in the Conan user home directory, e.g., *[CONAN_HOME]/global.conf*.
+
+Introduction to configuration
+-----------------------------
+
+*global.conf* is aimed to save some core/tools configuration variables that will be used by Conan. For instance:
```
Comment (czoido) [inline reference/config_files/global_conf.rst:11]: I think it's worth explaining what's the difference about the core config and the tools config

**Review by czoido — APPROVED** [review]
👏
Just one comment about explaining what a core config is.

File: reference/config_files/global_conf.rst (hunk 14 above)
Comment (franramirez688) [reply reference/config_files/global_conf.rst:11]: Yes! Agree 👍

**Pull request merged**